### PR TITLE
change position of the texts

### LINKED ISSUE #7: change position of the text
user should be able to drag the text on the image to place it wherever they want

### PR DESCRIPTION
**Pull request by harshN-17**
fixed #7 
added the functionality to drag the texts around the meme image and drop it anywhere on it.

### CODE REVIEW

File: src/components/Meme.js [hunk 1]
```diff
@@ -81,6 +67,77 @@ export default function Meme() {
...
+				}
+			}
+		]
+	)
+	
+	//redering the text data in h2 tag 
+	const memeTexts = memeTextData.map(text => {
+		return <h2
+				key={text.id}
+				draggable={text.draggable}
```
Comment (dev-lovedeep) [inline src/components/Meme.js:101]: draggable will always be true for all text, then why are you taking it from object

File: src/components/Meme.js [hunk 2]
```diff
@@ -81,6 +67,77 @@ export default function Meme() {
...
+					top: "0px",
+				}
+			},
+			{
+				id: 2,
+				draggable: true,
+				style: {
+					position: "absolute",
+					left: "50%",
+					transform: "translateX(-50%)",
```
Comment (dev-lovedeep) [inline src/components/Meme.js:89]: these properties are common, they must be in a class not in the object

File: src/components/Meme.js [hunk 3]
```diff
@@ -81,6 +67,77 @@ export default function Meme() {
...
+	const [memeTextData, setMemeTextData] = useState(
+		[
+			{
+				id: 1,
+				draggable: true,
+				style: {
+					position: "absolute",
+					left: "50%",
+					transform: "translateX(-50%)",
+					top: "0px",
```
Comment (dev-lovedeep) [inline src/components/Meme.js:80]: my suggestion would be to only store top and left in the object

File: src/components/Meme.js [hunk 4]
```diff
@@ -81,6 +67,77 @@ export default function Meme() {
...
+		e.preventDefault();
+		const id = e.dataTransfer.getData("id");
+		let x = e.clientX;
+		let y = e.clientY;
+		const rect = document.querySelector('.meme').getBoundingClientRect();
+		//create a temp text array 
+		const tempTexts = [...memeTextData]
+		//change the style of selected text and position it on the cursor
+		let index = tempTexts.findIndex((text) => text.id == id);
+		if(index == -1) return;
```
Comment (dev-lovedeep) [inline src/components/Meme.js:131]: use '==='

File: src/components/Meme.js [hunk 5]
```diff
@@ -81,6 +67,77 @@ export default function Meme() {
...
+
+	//when the text is dragged over the meme image
+	const handleDragOver = (e) => {
+		e.preventDefault();
+	}
+
+	//when the text is dropped on the meme image
+	const handleDrop = (e) => {
+		e.preventDefault();
+		const id = e.dataTransfer.getData("id");
```
Comment (dev-lovedeep) [inline src/components/Meme.js:123]: add comments to explain the logic of the drag and calculation

**Comment by harshN-17** [comment]
I have made the asked changes on the latest commit

**Pull request closed without merging**